src/input/game-pads.js

```diff
@@ -107,9 +107,31 @@ pc.extend(pc, function () {
 
             var i, len = pads.length;
             for (i = 0;i < len; i++) {
-                this.previous[i] = this.current[i];
-                this.current[i] = pads[i];
-            }            
+                // Initialize the previous and current for each gamepad
+                if(this.previous[i] == null){
```

Comment:
Please use === always 
And spacing of `if` statements must fit the coding standards doc: https://github.com/playcanvas/engine/blob/master/CONTRIB.md